### PR TITLE
React bindings

### PR DESCRIPTION
**Pull request by edoardocavazza**
# Introducing React bindings

This PR introduces build scripts to generate React bindings using the Stencil compiler.

----

### Motivation and Context

WebComponents and React are not best friends. React treats Custom Element nodes like any other child, making it impossible to pass objects and functions as references. In order to make them play nicely, the developer should create a React proxy that maps every property to the WebComponent.

DX is key for a Design System. Since many teams use React as main application framework, we need to make the integration more organic and painless.

In this PR we are going to extend the Stencil configuration in order to automatically generate React bindings.

### Priority

- [ ] 1 - Highest
- [x] 2 - High
- [ ] 3 - Medium
- [ ] 4 - Low
- [ ] 5 - Lowest
- [ ] 6 - Not a Priority

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Component (add a Component as approved dy Design System)
- [ ] Docs (add documentation)
- [ ] Chore (changes that adds small enhancement)
- [ ] Breaking (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by ufollettu** [comment]
Readme aside, do you think we need to update storybook docs to incorporate react bindings components?

**Comment by edoardocavazza** [comment]
In first instance, we need to fix the events documentation on the storybook. Then, if we find a convenient, automated way to document the bindings, we should certainly do that.

File: react/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  if (key in exports && exports[key] === library[key]) {
+    return;
+  }
+  Object.defineProperty(exports, key, {
+    enumerable: true,
+    get: function () {
+      return library[key];
+    },
+  });
+});
```
Comment (nitroin) [inline react/index.js:22]: Stavo cercando di capire se possiamo fare a meno di questo.

Peró non ho chiaro i passaggi:
- https://github.com/ionic-team/stencil-ds-output-targets#react
- https://github.com/ionic-team/stencil-ds-react-template

Comment (edoardocavazza) [reply react/index.js:22]: Era stato messo per avere una shortcut ai dist di React. Finché era la versione con `export * from './index.js';` funzionava abbastanza, ma Jest non la supporta, e così effettivamente è abbastanza brutto.
Tre strade direi:
1. `'@zanichelli/albe-web-components/react'` --> `'@zanichelli/albe-web-components/react/dist'`
2. Provare a fare la dist da `react/src` dentro a `react`, ma non mi piace molto....
3. Usare la cartella react come output per Stencil e fare il transpiling typescript in place (via di mezzo tra 1 e 2)

**Review by a-sartori-wt — APPROVED** [review]

**Review by nitroin — APPROVED** [review]

**Review by ufollettu — APPROVED** [review]

**Pull request merged**